test/gtest/stringify.cpp

```diff
@@ -0,0 +1,127 @@
+#include "ir/utils.h"
+#include "passes/stringify-walker.h"
+#include "print-test.h"
+#include "wasm.h"
+#include "gtest/gtest.h"
+
+using namespace wasm;
+
+using StringifyTest = PrintTest;
+
+TEST_F(StringifyTest, Print) {
+  auto moduleText = R"wasm(
+    (module
+    (tag $catch_a (param i32))
+    (tag $catch_b (param i32))
+    (tag $catch_c (param i32))
+    (func $d
+      (block $block_a
+        (drop (i32.const 20))
+        (drop (i32.const 10))
+      )
+      (block $block_b
+        (drop (if (i32.const 0)
+          (i32.const 40)
+          (i32.const 5)
+        ))
+      )
+      (block $block_c
+        (drop (if (i32.const 1)
+          (i32.const 30)
+        ))
+      )
+      (block $block_d
+        (try $try_a
+          (do
+            (nop)
+          )
+          (catch $catch_a
+            (drop (i32.const 8))
+          )
+          (catch $catch_b
+            (drop (i32.const 15))
+          )
+        )
+      )
+      (block $block_e
+        (try $try_b
+          (do
+            (nop)
+          )
+          (catch $catch_c
+            (drop (i32.const 33))
+          )
+        )
+      )
+    )
+   )
+  )wasm";
+
+  auto stringifyText = R"stringify(in visitExpression for block
+adding unique symbol
+in visitExpression for block $block_a
```

Comment:
Wait, shouldn't we visit the contents of `block_a` before continuing to b? Maybe I'm not understanding the example from the headers. Looking at that example again, it has nested control flow but not sibling control flow - maybe expanding the example to include that could avoid confusion.

Reply:
I updated the comments and added a sibling control flow to the contrived wat example. Hopefully this makes the explanation a bit more clear. Please let me know if not!

This is the correct traversal. 
We want to visit the function body,
then the children of the function body as a single unit: block_a, block_b, block_c, block_d
and then visit the contents of each block. 

If we run into further control flow, we defer visiting the contents of the control flow until after all siblings are visited.

Reply:
I think I see now, thanks! So this doesn't keep a control flow structure close to its children. I was assuming that in my head for some reason but I guess it's not the goal here.